client/jetpack-connect/plans.jsx

```diff
@@ -58,30 +58,30 @@ class Plans extends Component {
 		}
 	}
 
-	componentWillReceiveProps = nextProps => {
+	componentDidUpdate() {
 		if ( ! this.redirecting ) {
-			this.maybeRedirect( nextProps );
+			this.maybeRedirect();
 		}
-	};
+	}
 
-	maybeRedirect = props => {
-		if ( props.isAutomatedTransfer ) {
-			this.props.goBackToWpAdmin( props.selectedSite.URL + JETPACK_ADMIN_PATH );
+	maybeRedirect() {
+		if ( this.props.isAutomatedTransfer ) {
+			this.props.goBackToWpAdmin( this.props.selectedSite.URL + JETPACK_ADMIN_PATH );
 		}
-		if ( props.selectedPlan ) {
-			this.selectPlan( props.selectedPlan );
+		if ( this.props.selectedPlan ) {
+			this.selectPlan( this.props.selectedPlan );
 		}
-		if ( props.hasPlan || props.notJetpack ) {
+		if ( this.props.hasPlan || this.props.notJetpack ) {
 			this.redirect( CALYPSO_PLANS_PAGE );
 		}
-		if ( ! props.canPurchasePlans ) {
-			if ( props.isCalypsoStartedConnection ) {
+		if ( ! this.props.canPurchasePlans ) {
+			if ( this.props.isCalypsoStartedConnection ) {
 				this.redirect( CALYPSO_REDIRECTION_PAGE );
 			} else {
-				this.redirectToWpAdmin( props );
+				this.redirectToWpAdmin( this.props );
```

Comment:
`redirectToWpAdmin` no longer takes props as an argument, should we remove the argument from here? Note it's used in `selectFreeJetpackPlan` too.